activesupport/lib/active_support/i18n_railtie.rb

```diff
@@ -63,7 +63,8 @@ def self.initialize_i18n(app)
       if app.config.reloading_enabled?
         directories = watched_dirs_with_extensions(reloadable_paths)
         reloader = app.config.file_watcher.new(I18n.load_path.dup, directories) do
-          I18n.load_path.keep_if { |p| File.exist?(p) }
+          # Do not watch translations loaded via gems.
+          I18n.load_path.keep_if { |p| p.start_with?(Rails.root.to_s) && File.exist?(p) }
```

Comment:
So I think pruning gem translations from the load path entirely makes them inaccessible when reloading. The tests failing seem to support that. Maybe we could look at doing something like this instead?
```rb
I18n.load_path.delete_if { |p| p.start_with?(Rails.root.to_s) && !File.exist?(p) }
```
(that way we're doing less `exist?` checks)

And also having the watcher watch all of `I18n.load_path` might not make sense if it has gem translations, so we could try to filter above.

Reply:
> So I think pruning gem translations from the load path entirely makes them inaccessible when reloading. 
Yes. It seems now there are unintended consequences to removing elements from that collection.

I now filter at the watcher initialization. For both [direct files](https://github.com/rails/rails/pull/52271/files#diff-58bae95612ce1b0046ec9b1e7bbe6fb04ae263ea8245cc528ccfd849361b88aeR79-R81) and `directories_to_watch`.

Since activesupport is a standalone gem I was extra cautious and only perform the filter if `Rails.root` is defined. What do you think? 

`@files.keep_if { |file| file.to_s.start_with?(Rails.root.to_s) } if defined?(Rails) && Rails.respond_to?(:root)`

The only complaints I now see on CI have to do with a deprecated `:unprocessble_entity` against Rack#head; and a failed request for the changelog. So I think this is green.

Reply:
> Since activesupport is a standalone gem I was extra cautious and only perform the filter if Rails.root is defined.

This is a railtie, so its code will only be executed if Rails is defined. I don't think you need to be cautious.

Reply:
> This is a railtie, so its code will only be executed if Rails is defined. 
Ah yes of course, sorry 😅 

> Can't we just filter the files array we pass to the initializer instead of editing the watcher class?

Much simpler, yes. I now filter within Rails.root to the collection [we pass into the watcher](https://github.com/rails/rails/pull/52271/files) at initialization.

[> I18n.load_path.delete_if { |p| p.start_with?(Rails.root.to_s) && !File.exist?(p) }](https://github.com/rails/rails/pull/52271/files#diff-ac85f70aba8d19a99934ca1adfa24a0fcad978ab27880f5248bd36230e37ce70R66)

During bootup this still carries a fair bit of weight performance-wise with all the `File.exist?` checks. I'm wondering what this helps with. The case where someone has deleted filed with a server running? Incorrectly specified `l18n.load_path`s? I'm just weighing the perf impact 100% of the time on boot vs. the performance win once files that don't exist are pruned.

We can keep it but just making a note while I'm here.